texar/utils/beam_search.py

```diff
@@ -16,7 +16,10 @@
 # ==============================================================================
 """
 Implementation of beam search with penalties.
-Adapted from tensor2tensor repository.
+
+Adapted from:
+    `https://github.com/tensorflow/tensor2tensor/`
+    `blob/master/tensor2tensor/utils/beam_search.py`
```

Comment:
URLs should not be broken into multiple lines. They are skipped during line length checks and spelling checks.

Reply:
And I agree with @haoransh on adding a permalink.